pgrx-examples/hooks/src/lib.rs

```diff
@@ -0,0 +1,225 @@
+use pgrx::prelude::*;
+
+::pgrx::pg_module_magic!(name, version);
+
+fn delete_must_have_a_where(query: PgBox<pg_sys::Query>) {
+    if query.commandType != pg_sys::CmdType::CMD_DELETE {
+        return ();
+    }
+
+    // SAFETY: the jointree is always defined in a DELETE Query
+    let jointree = unsafe { PgBox::from_pg(query.jointree) };
+
+    if jointree.quals.is_null() {
+        panic!("DELETE queries must have a WHERE clause");
+    }
+}
+
+fn only_superusers_can_truncate(pstmt: PgBox<pg_sys::PlannedStmt>) {
+    if !unsafe { pgrx::is_a(pstmt.utilityStmt, pg_sys::NodeTag::T_TruncateStmt) } {
+        return ();
+    }
+
+    if !unsafe { pg_sys::superuser() } {
+        panic!("Only superusers can truncate")
+    }
+}
+
+unsafe fn register_hooks() {
+    //
+    // Post Parse Analyze hook
+    //
+    static mut PREV_POST_PARSE_ANALYZE_HOOK: pg_sys::post_parse_analyze_hook_type = None;
+    PREV_POST_PARSE_ANALYZE_HOOK = pg_sys::post_parse_analyze_hook;
+    pg_sys::post_parse_analyze_hook = Some(post_parse_analyze_hook);
```

Comment:
❓ Is it reasonable to use `replace` here?


```suggestion
    PREV_POST_PARSE_ANALYZE_HOOK =
        pg_sys::post_parse_analyze_hook.replace(post_parse_analyze_hook);
```

Reply:
@cbandy : it produces the following warning

```rust

warning: creating a mutable reference to mutable static
  --> src/lib.rs:77:5
   |
77 |     pg_sys::ProcessUtility_hook.replace(process_utility_hook);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable reference to mutable static
   |

Reply:
Thanks for trying. I see the same here.

I _don't_ get that in my extension using `emit_log_hook` with `#![deny(static_mut_refs)]` and `#![warn(unsafe_op_in_unsafe_fn)]`. 😩

```rust
    static mut HOOK_EMIT_LOG: pg_sys::emit_log_hook_type = None;
    HOOK_EMIT_LOG = pg_sys::emit_log_hook.replace(my_emit_log_hook);
```